From b09ff9d4a27c8f0fff7b8156c2320d2262e9d796 Mon Sep 17 00:00:00 2001
From: Hanjie Lin <hanjie.lin@amlogic.com>
Date: Tue, 8 Nov 2022 19:39:49 +0800
Subject: [PATCH 69/95] module: add ignore_check_version [1/1]

PD#SWPL-99365

Problem:
need allow disagree ko continue to insmod default

Solution:
add ignore_check_version default to 1, and print warning
when insmod disagree ko

Verify:
sc2

Change-Id: I6328b20ae582f0708ce157b218faa90b9603d65b
Signed-off-by: Hanjie Lin <hanjie.lin@amlogic.com>
---
 kernel/module.c | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/kernel/module.c b/kernel/module.c
index 3a0069d211f6..d41057b8dbb9 100644
--- a/kernel/module.c
+++ b/kernel/module.c
@@ -1252,6 +1252,11 @@ static u32 resolve_rel_crc(const s32 *crc)
 	return *(u32 *)((void *)crc + *crc);
 }
 
+#if IS_ENABLED(CONFIG_AMLOGIC_DEBUG)
+static int ignore_check_version = 1;
+core_param(ignore_check_version, ignore_check_version, int, 0644);
+#endif
+
 static int check_version(const struct load_info *info,
 			 const char *symname,
 			 struct module *mod,
@@ -1298,7 +1303,14 @@ static int check_version(const struct load_info *info,
 bad_version:
 	pr_warn("%s: disagrees about version of symbol %s\n",
 	       info->name, symname);
+#if IS_ENABLED(CONFIG_AMLOGIC_DEBUG)
+	pr_warn("!!!MUST FIX!!! %s: ko need recompile.\n", info->name);
+	dump_stack();
+	return ignore_check_version;
+#else
 	return 0;
+#endif
+
 }
 
 static inline int check_modstruct_version(const struct load_info *info,
-- 
2.25.1

